src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolverInterface.php

```diff
@@ -32,15 +32,17 @@
     public function supports(Request $request, ArgumentMetadata $argument);
 
     /**
-     * Get the actual value for the argument.
+     * Yield the possible value(s).
      *
-     * The return must always be an array:
-     *  - return array($myArgumentValue);
+     * One yield is equal to one argument, only yield more than once if you're working with
+     * variadic arguments or something similar.
```

Comment:
I would be more generic here and say that an implementation must yield at least one value.